docs/configuration/flag_configuration.md

```diff
@@ -0,0 +1,126 @@
+
+Example:
+
+```
+"variants": {
+  "red": "c05543",
+  "green": "2f5230",
+  "blue": "0d507b"
+}
+```
+
+Example:
+
+```
+"variants": {
+  "on": true,
+  "off": false
+}
+```
+
+Example of an invalid configuration:
+
+```
+"variants": {
+  "on": true,
+  "off": "false"
+}
+```
+
+### Default Variant
+
+`defaultVariant` is a **required** property. The value **must** match the name of one of the variants defined above. The default variant is always used unless a targeting rule explicitly overrides it.
+
+Example:
+
+```
+"variants": {
+  "on": true,
+  "off": false
+},
+"defaultVariant": "off"
+```
+
+Example:
+
+```
+"variants": {
+  "red": "c05543",
+  "green": "2f5230",
+  "blue": "0d507b"
+},
+"defaultVariant": "red"
+```
+
+Example of an invalid configuration:
+
+```
+"variants": {
+  "red": "c05543",
+  "green": "2f5230",
+  "blue": "0d507b"
+},
+"defaultVariant": "purple"
+```
+
+### Targeting Rules
+
+`targeting` is an **optional** property. A targeting rule **must** be valid JSON. Flagd uses a modified version of [JSON Logic](https://jsonlogic.com/), as well as some custom pre-processing, to evaluate these rules. The output of the targeting rule **must** match the name of one of the variants defined above. If an invalid or null value is is returned by the targeting rule, the `defaultVariant` value is used.
+
+The [JSON Logic playground](https://jsonlogic.com/play.html) is a great way to experiment with new targeting rules. The following example shows how a rule could be configured to return `binet` when the email (which comes from evaluation context) contains `@faas.com`. If the email wasn't included in the evaluation context or doesn't contain `@faas.com`, null is returned and the `defaultVariant` is used instead.
+
+<details>
+  <summary>Click here to see how this targeting rule would look in the JSON Logic playground.</summary>
+
+1. Open the [JSON Logic playground](https://jsonlogic.com/play.html) in your favorite browser
+1. Add the follow JSON as the `Rule`:
+
+    ```json
+    {
+      "if": [
+        {
+          "in": [
+            "@faas.com",
+            {
+              "var": ["email"]
+            }
+          ]
+        },
+        "binet",
+        null
+      ]
+    }
+    ```
+
+1. Add the following JSON as the `Data`:
+
+    ```json
+    {
+      "email": "test@faas.com"
+    }
+    ```
+
+1. Click `Compute`
+1. confirm the output show `"binet"`
+1. Optionally, experiment with different rules and data
```

Comment:
You may want to mention something about the `TARGETING_MATCH` reason, vs `DEFAULT` and `STATIC`.

Reply:
Not sure if it's too much, but we could also possibly mention implications on caching here, but also we could just link: https://github.com/open-feature/flagd/blob/5830592053c55dc9e55c16854e40c3fc8345d6d1/docs/caching.md

Reply:
I've added a small reference and a link, it may be worthwhile adding a section for response types? what do you think

Reply:
I think this is sufficient for now.